### DISCUSSION TITLE
[Detector Support]: Cannot create yolonas model

### DISCUSSION THREAD

**Discussion by chrasim1** [blakeblackshear/frigate#21122] [Detector Support]
### Describe the problem you are having

I'm trying to generate the Yolonas ONNX models with the Google Colab Notebook.

However, in the model.export step, I get an error message that onnxscript is missing. When I try to install onnxscript, it requires updating the onnx version which does not meet the requirements of super.gradients.

How can this be fixed?

### Version

0.16-2

### Frigate config file

```yaml
n/a
```

### docker-compose file or Docker CLI command

```yaml
n/a
```

### Relevant Frigate log output

```shell
n/a
```

### Install method

Docker Compose

### Object Detector

OpenVino

### Screenshots of the Frigate UI's System metrics pages

n/a

**Comment by inkeep-ai-assistant** [comment]
The dependency conflict you're experiencing with the YOLO-NAS export notebook is a known issue[(1)](https://github.com/blakeblackshear/frigate/discussions/16702). The problem occurs because `super_gradients==3.7.1` has conflicting dependencies with the required ONNX packages[(1)](https://github.com/blakeblackshear/frigate/discussions/16702).

This is not a Frigate bug but rather a dependency conflict in the upstream libraries[(1)](https://github.com/blakeblackshear/frigate/discussions/16702). The issue stems from `super_gradients==3.7.1` requiring `numpy<=1.23`, while `onnxruntime 1.15.0` requires `numpy>=1.24.2`[(1)](https://github.com/blakeblackshear/frigate/discussions/16702).

Unfortunately, there isn't a straightforward fix documented in the available sources for this specific dependency conflict. The community discussion acknowledges this as an external dependency issue rather than something Frigate can directly resolve[(1)](https://github.com/blakeblackshear/frigate/discussions/16702).

**Alternative approaches you might consider:**

1. Try using a different Python environment or Google Colab runtime that might have compatible package versions
2. Experiment with loosening package version constraints if you're comfortable modifying the notebook requirements
3. Consider using pre-trained YOLO-NAS models that others have successfully exported, if available in the community

Since this is an upstream dependency issue with the model export process rather than Frigate itself, you may want to check the super-gradients GitHub repository for any updates or workarounds to the numpy version conflict.

📚 **Sources:**
1. [[Support]: Problem with super_gradients==3.7.1 in  /notebooks/YOLO_NAS_Pretrained_Export.ipynb](https://github.com/blakeblackshear/frigate/discussions/16702)

---

Have another question? Just tag [@inkeep](https://inkeep.com).

**Reply by chrasim1** [reply]
The AI answer is referring to an older issue, not to the one I'm facing.

**Reply by hawkeye217** [reply]
Looks like package versions may need to be pinned in order to get the notebook to work again. 

Depending on your object detector, you could use a YOLOv9 model instead of YOLO-NAS: https://docs.frigate.video/configuration/object_detectors/#yolov9

**Reply by chrasim1** [reply]
I tried to pin package versions, but found that the only version of onnxscript compatible with onnx-1.1.5 is an old 0.10.-prerelease, and using that one will give an error in step 4. This issue seems to have come with Python >3.10, maybe I'll try a local installation with an older python environment. Or maybe someone could send me the yolo-nas-s and yolo-nas-m onnx models (320x320) to chrasim@gmail.com?

**Comment by Trteich** [comment]
To fix it appeared to be extreamly easy

1. Change environment to 2025.07(top menu -> Runtime -> Change runtime type)
2. Add code block before the very first one
```
!pip install jedi>=0.16
```
3. Change from python3.12 to python3.11 in code block
```
!sed -i 's/sghub.deci.ai/://sg-hub-nv.s3.amazonaws.com' /usr/local/lib/python3.12/dist-packages/super_gradients/training/pretrained_models.py
!sed -i 's/sghub.deci.ai/://sg-hub-nv.s3.amazonaws.com' /usr/local/lib/python3.12/dist-packages/super_gradients/training/utils/checkpoint_utils.py
```
3. tap Run all
